app/models/product.rb

```diff
@@ -2,20 +2,26 @@ class Product < ApplicationRecord
   has_many :line_items
   has_many :orders, through: :line_items
 
-  validates :title, :description, :image_url, presence: { message: "can't be blank"}
+  validates :url, presence: true, url: true
+  validates :title, :description, :permalink, presence: true
   validates :title, uniqueness: true
```

Comment:
this should also be in with_options block

Reply:
done.